ratbag_emu/endpoint.py

```diff
@@ -0,0 +1,103 @@
+# SPDX-License-Identifier: MIT
+
+import logging
+import struct
+import typing
+
+import hidtools.uhid
+
+from typing import List
+
+if typing.TYPE_CHECKING:
+    from ratbag_emu.device import Device  # pragma: no cover
+
+
+class Endpoint(hidtools.uhid.UHIDDevice):
+    '''
+    Represents a device endpoint
+
+    A HID device is created for each endpoint. The enpoint can be used to
+    receive and send data
+
+    :param owner:   Endpoint owner
+    :param rdesc:   Report descriptor
+    :param number:  Endpoint number
+    '''
+
+    def __init__(self, owner: 'Device', rdesc: List[int], number: int):
+        super().__init__()
+
+        self.__logger = logging.getLogger('ratbag-emu.endpoint')
+
+        self._owner = owner
+
+        self._info = owner.info
+        self.rdesc = rdesc
+        self.number = number
+        self.name = f'ratbag-emu {owner.id} ({owner.name}, ' \
+                    f'{self.vid:04x}:{self.pid:04x}, {self.number})'
+
+        self._output_report = self._receive
+
+        self.create_kernel_device()
```

Comment:
nitpick (for a future use): you might want to split the creation of the object and the actual registering to uhid  which will create the kernel virtual device. This is useful when you want to first get your object, check for some properties on it (like which min/max do we have), and then skip a test.

Reply:
Noted, thanks.